--- /home/quan/Works/Git/phimp.me.webserver/additional_modules/views_galleriffic/views_galleriffic.module
+++ /home/quan/Works/drupal-7.12/sites/all/modules/views_galleriffic/views_galleriffic.module
@@ -76,6 +76,8 @@
   $view = $vars['view'];
   $options = $vars['options'];
   $vars['fields'] = array(); // ensure it's at least an empty array.
+  global $user;//get current user
+  $uid = 0;//init $uid id author
   foreach ($view->field as $id => $field) {
     $field_output = $view->field[$id]->theme($vars['row']);
     $object = array();
@@ -108,6 +110,17 @@
           $vars['fields']['slide']->content = $slide_url['2'];
         }
       }
+    // Init uid_field and edit fied
+    if ($object['option'] == 'uid_field') {
+      $uid = check_markup($vars['fields']['uid_field']->content, $format_id = 'full_html', '', $check = FALSE);
+    }
+    //get number $uid
+    $uid = str_replace("<p>", "", $uid);
+    $uid = str_replace("</p>", "", $uid);
+       if (($object['option'] == 'edit_field') && (intval($uid) == $user->uid || $user->uid == 1)) {
+         $vars['fields']['edit']->content = check_markup($vars['fields']['edit_field']->content, $format_id = 'full_html', '', $check = FALSE);
+       }
+    else $vars['fields']['edit']->content = "";
     }
   }
   // Allow for empty fields that didn't already get defined.
